tools/update.php

```diff
@@ -0,0 +1,746 @@
+ * instance to the newest install without issue.
+ *
+ * This includes having the latest minor release files, third-party packages
+ * (apt, npm, composer), PHP Version, etc.
+ *
+ * ## Details
+ *
+ * Specifically:
+ *
+ *- [x] **Creates a back up of `$LORIS_root`**
+ *
+ *- [x] (If not on a development instance) **Downloads the release source code
+ *      and overwrites `$LORIS_root`**
+ *
+ *- [x] **Ensures that apt dependencies are installed and up-to-date**
+ *
+ *- [x] **Runs 3rd-party package managers** (npm and composer)
+ *
+ *- [x] **Outputs a list of SQL patches** for the user to apply manually
+ *      (including development patches if on a LORIS dev instance)
+ *
+ *- [x] **Provides a lot of output** to assist the user in what's going on.
+ *
+ * Assumptions:
+ *
+ * As of now there is no "rollback" feature. `rsync` is used to overwrite the
+ * LORIS-root source code from the unzipped release folder. A backup is created,
+ * but should something go wrong with `rsync`, the user will need to restore the
+ * source code files manually. In this case, an error message is displayed and
+ * the script will exit.
+ *
+ * The function to retrieve patches is heavily dependent on the way we organize
+ * and name SQL files. If the directory structure changes, this functionality
+ * is likely to break.
+ *
+ * @category Update
+ * @package  Tools
+ * @author   John Saigle <john.saigle@mcgill.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+
+require_once __DIR__ . "/../vendor/autoload.php";
+
+error_reporting(E_ALL);
+$required_major_php = 7; //TODO: Update these values as time passes
+$required_minor_php = 2;
+// PHP version required for LORIS.
+$php_version = "$required_major_php.$required_minor_php";
+if (PHP_MAJOR_VERSION < $required_major_php
+    || PHP_MINOR_VERSION < $required_minor_php
+) {
+    die("[-] ERROR: {$argv[0]} and LORIS require PHP v$php_version or higher.");
+}
+
+/* Create db connection and get version info. */
+$config    = \NDB_Config::singleton();
+$db_config = $config->getSetting('database');
+$db        = \Database::singleton(
+    $db_config['database'],
+    $db_config['username'],
+    $db_config['password'],
+    $db_config['host']
+);
+$paths     = $config->getSetting('paths');
+$loris_root_dir = $paths['base'];
+
+$preupdate_version = getVersionFromLORISRoot($loris_root_dir);
+$info            = json_decode(getLatestReleaseInfo());
+$release_version = $info->{'tag_name'};
+$release_version = substr($release_version, 1); // remove leading 'v'
+
+// Check that the backup directory argument is present and valid
+$backup_dir = $argv[1] ?? '';
+if (!is_dir($backup_dir)) {
+    echo '[-] ERROR: Argument suppled for backup directory is not a valid '
+        . 'directory.' . PHP_EOL;
+    die(usageString());
+}
+// Only apply SQL patches if user explicitly requests it
+$apply_patches = false;
+if (in_array('--apply-patches', $argv)) {
```

Comment:
always better to use `true` as 3rd arg when using `in_array()`

see http://php.net/manual/en/function.in-array.php#106319

Reply:
same for all below